### PR TITLE
Make nodeless setup even easier and add some docs

### PR DESCRIPTION
**Pull request by timebertt**
**What this PR does / why we need it**:
This PR improves the nodeless local setup to make it even easier to get started with it.
Now you can just run `make local-garden-up` and `make dev-setup` and your good to go.
I also added some documentation for how to get started with in `docs/development/local_setup.md`.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```improvement developer
It is now even easier to get started developing on Gardener by leveraging the nodeless local setup. Please refer to the [documentation](https://github.com/gardener/gardener/blob/master/docs/development/local_setup.md) to see how to use it.
```

### CODE REVIEW

File: hack/dev-setup [hunk 1]
```diff
@@ -214,4 +219,56 @@ OaMnRwirobPupceHD5zvuGPjjDaPhc1yeRMYo+xXgLL+mYyXR1ZYGuI=
 -----END RSA PRIVATE KEY-----
 EOF
 
+cat <<EOF > "${DEV_DIR}/tls/gardener-apiserver.crt"
```
Comment (zanetworker) [inline hack/dev-setup]: Why is this needed? isn't it enough to fetch it like we do in the `start` scripts?

File: hack/dev-setup [hunk 2]
```diff
@@ -214,4 +219,56 @@ OaMnRwirobPupceHD5zvuGPjjDaPhc1yeRMYo+xXgLL+mYyXR1ZYGuI=
...
+-----BEGIN CERTIFICATE-----
```
Comment (zanetworker) [inline hack/dev-setup]: Why is this needed?

File: hack/dev-setup [hunk 3]
```diff
@@ -214,4 +219,56 @@ OaMnRwirobPupceHD5zvuGPjjDaPhc1yeRMYo+xXgLL+mYyXR1ZYGuI=
...
+EcAWF3fLNghoKA1K3xElk7HV1NsshAUJ0bZZI5UWHVLHZ2sqxCIsBaI0J8rvwXWm
+zvTCpYdnDKfbg/DoPTrFxtjRgR4fwUImfOmm6PkvO7r+549iTKuTM5X74jiH6Btu
+rd3a48Ob+z1+uYJaz4dJBxAO0uKR2WNOE+GP00/7/BzMoE6m/71I+DUb7qZGJXof
+uye5YGW7sy0Rg/d1pJaUQSwZiXzG8C+lB8qTeE99CLzu3D484Jn6hIPUgPTHEpf3
+HAkYEHrDOQo2d81QraDYvI2MDCUL41LbmIL4zP6NZK9z93KGQD1bM3oAt5MJMspN
+nCf37qW2nh9JjiJ1m8hb
+-----END CERTIFICATE-----
+EOF
+
+cat <<EOF > "${DEV_DIR}/tls/gardener-apiserver.key"
```
Comment (zanetworker) [inline hack/dev-setup]: Why is this needed? isn't it enough to fetch it like we do in the start scripts?

File: hack/dev-setup (hunk 1 above)
Comment (timebertt) [reply hack/dev-setup]: I added these files to be able to use them inside my debug configurations (I cannot fetch the certificates like in the `start` scripts there).
But you are right, this is not needed. I can simply use paths to the files already present inside `hack/local-garden`, like `hack/local-garden/certificates/certs/default-admin.crt` for my debug configs.
So it removed it from `hack/dev-setup`, PTAL :)

**Review by zanetworker — DISMISSED** [review]
/lgtm

**Review by timuthy — DISMISSED** [review]

**Comment by timuthy** [comment]
@tim-ebert can you force push to wake up license/cla?

**Comment by timebertt** [comment]
Done

**Review by timuthy — APPROVED** [review]
